src/components/Button.tsx

```diff
@@ -38,6 +40,38 @@ export function ButtonRoot({
   ] as unknown as HtmlEscapedString
 }
 
+export type ButtonAddActionProps = ButtonProps & {
+  /** URL of the action. */
+  url: string
```

Comment:
We should keep consistent with other buttons and use `action`. If `action` detects an absolute URL, then it will use it.

```suggestion
        action="/log-this"
```

Reply:
> We should keep consistent with other buttons and use `action`. If `action` detects an absolute URL, then it will use it.

I get what you're saying, but the URL that's passed here must be absolute, as it later goes as a value to the `postUrl` parameter in the deeplink.

In the suggested changes you've provided (as far as I understand to be an example of usage of such), such would only work if the `ButtonAddAction` would somehow would be able to construct an absolute URL out of this – potentially knowing full routing tree (since multiple frogs could be joined via `app.route()`) to get the current route with all the base paths and `c.origin` to prepend the url with it.

Reply:
Maybe I'm missing out something.

Reply:
Now I think that we could handle this in `parseIntent` method.

Reply:
If that's possible, there would be no need to expose `origin` to the `Context`.

Investigating and most likely think it should be.

Reply:
yep